server/Arcadia.Assistant.CSP/Vacations/VacationsEmailLoader.cs

```diff
@@ -0,0 +1,222 @@
+                                    case GetInboxEmails.Success msg:
+                                        return new LoadInitialState.Success(msg.Emails);
+
+                                    case GetInboxEmails.Error msg:
+                                        return new LoadInitialState.Error(msg.Exception);
+
+                                    default:
+                                        return new LoadInitialState.Error(
+                                            new Exception("Unexpected inbox emails response"));
+                                }
+                            },
+                            failure: err => new LoadInitialState.Error(err)
+                        );
+                    break;
+
+                case LoadInitialState.Success msg:
+                    this.LoadVacationsFromEmails(msg.Emails);
+                    this.BecomeAfterInitial();
+                    break;
+
+                case LoadInitialState.Error msg:
+                    this.logger.Warning(msg.Exception.Message);
+                    this.BecomeAfterInitial();
+                    break;
+
+                default:
+                    this.Stash.Stash();
+                    break;
+            }
+        }
+
+        private void BecomeAfterInitial()
+        {
+            this.Stash.UnstashAll();
+            Context.System.EventStream.Subscribe<InboxEmailsEventBus>(this.Self);
+            this.Become(this.AfterInitialState);
+        }
+
+        private void AfterInitialState(object message)
+        {
+            switch (message)
+            {
+                case GetVacationsInfo _:
+                    var result = new GetVacationsInfo.Success(this.vacationsInfo);
+                    this.Sender.Tell(result);
+                    break;
+
+                case InboxEmailsEventBus msg:
+                    this.LoadVacationsFromEmails(msg.Emails);
+                    break;
+
+                default:
+                    this.Unhandled(message);
+                    break;
+            }
+        }
+
+        private void LoadVacationsFromEmails(IEnumerable<Email> emails)
+        {
+            var lastActualEmail = emails
+                .Where(this.CheckEmail)
+                .OrderByDescending(e => e.Date)
+                .FirstOrDefault();
+            if (lastActualEmail == null)
+            {
+                return;
+            }
+
+            this.logger.Debug(
+                $"Received vacations email from {lastActualEmail.Sender} with subject {lastActualEmail.Subject}"
+            );
+
+            var vacationsAttachment = Encoding.UTF8.GetString(lastActualEmail.Attachments.First());
+            this.vacationsInfo = this.ParseVacations(vacationsAttachment);
+        }
+
+        private List<EmployeeVacationRecord> ParseVacations(string vacationsAttachment)
+        {
+            var lines = vacationsAttachment.Split(new[] { "\r\n" }, StringSplitOptions.None);
```

Comment:
-> `new[] {"\r\n", "\n" }` to support weird environment scenarios

Reply:
Done.